draft-ietf-ppm-dap.md

```diff
@@ -1267,106 +1322,168 @@ This message consists of:
+
+1. Else the type is invalid, in which case the Leader MUST abort the
+   aggregation job.
+
+(Note: Since VDAF preparation completes in a constant number of rounds, it will
+never be the case that some reports are completed and others are not.)
 
 #### Helper Initialization {#aggregation-helper-init}
 
-Each Helper begins their portion of the aggregate initialization when they
-receive an AggregationJobInitReq message from the Leader. For each ReportShare
-conveyed by this message, the Helper attempts to initialize VDAF preparation
-(see {{Section 5.1 of !VDAF}}) just as the Leader does. If successful, it
-includes its prepare message in its response that the Leader will use to
-continue the process.
+The Helper begins an aggregation job when it receives an `AggregationJobInitReq`
+message from the Leader. For each `PrepareInit` conveyed by this message, the
+Helper attempts to initialize VDAF preparation (see {{Section 5.1 of !VDAF}})
+just as the Leader does. If successful, it includes the result in its response
+that the Leader will use to continue preparing the report.
+
+To begin this process, the Helper checks if it recognizes the task ID. If not,
+then it MUST abort with error `unrecognizedTask`.
 
-To begin this process, the Helper first checks if it recognizes the task ID. If
-not, then it MUST abort with error `unrecognizedTask`. Then the Helper checks
-that the report IDs in `AggregationJobInitReq.report_shares` are all distinct.
-If two ReportShare values have the same report ID, then the helper MUST abort
-with error `invalidMessage`. If this check succeeds, the helper then attempts to
-recover each input share in `AggregationJobInitReq.report_shares` as follows:
+Next, the Helper checks that the report IDs in
+`AggregationJobInitReq.prepare_inits` are all distinct. If two preparation
+initialization messages have the same report ID, then the Helper MUST abort with
+error `invalidMessage`.
+
+The Helper is now ready to process each report share into an outbound prepare
+step to return to the server. First the Helper preprocesses each report as
+follows:
 
 1. Decrypt the input share for each report share as described in
    {{input-share-decryption}}.
 1. Check that the resulting input share is valid as described in
    {{input-share-validation}}.
-1. Initialize VDAF preparation and initial outputs as described in
-   {{input-share-prep}}.
 
-Once the Helper has processed each report share in
-`AggregationJobInitReq.report_shares`, the Helper creates an
-`AggregationJobResp` message to complete its initialization. This message is
-structured as follows:
+For any report that was rejected, the Helper sets the outbound prep step to
 
 ~~~
-enum {
-  continued(0),
-  finished(1),
-  failed(2),
-  (255)
-} PrepareStepState;
+struct {
+  ReportId report_id;
+  PrepareStepState prepare_step_state = reject;
+  ReportShareError report_share_error;
+} PrepareStep;
```

Comment:
Now that `PrepareStep` is never sent by the Leader by the Helper, perhaps it deserves a more distinguish name, like `PrepareResp` or something?